tasks/src/main/java/com/example/Quickstart.java

```diff
@@ -0,0 +1,173 @@
+      .build();
+
+  public static void main(String... args) throws Exception {
+    Options options = new Options();
+    options.addOption(PROJECT_ID_OPTION);
+    options.addOption(QUEUE_OPTION);
+    options.addOption(LOCATION_OPTION);
+
+    if (args.length == 0) {
+      printUsage(options);
+      return;
+    }
+
+    CommandLineParser parser = new DefaultParser();
+    CommandLine params;
+    try {
+      params = parser.parse(options, args);
+    } catch (ParseException e) {
+      System.err.println("Invalid command line: " + e.getMessage());
+      printUsage(options);
+      return;
+    }
+
+    String projectId;
+    if (params.hasOption("project-id")) {
+      projectId = params.getOptionValue("project-id");
+    } else {
+      projectId = System.getenv(GGOGLE_CLOUD_PROJECT_KEY);
+    }
+    if (Strings.isNullOrEmpty(projectId)) {
+      printUsage(options);
+      return;
+    }
+
+    String queue = params.getOptionValue(QUEUE_OPTION.getOpt());
+    String location = params.getOptionValue(LOCATION_OPTION.getOpt());
+
+    switch (args[0]) {
+      default:
+        printUsage(options);
+        break;
+      case "create-task":
+        createTask(projectId, queue, location);
+        break;
+      case "lease-and-ack-task":
+        pullAndAckTask(projectId, queue, location);
+        break;
+    }
+  }
+
+  // [START cloud_tasks_create_task]
+  private static void createTask(String projectId, String queueName, String location)
+      throws IOException {
+    try (CloudTasksClient client = CloudTasksClient.create()) {
+      Task.Builder taskBuilder = Task
+          .newBuilder()
+          .setPullMessage(PullMessage.newBuilder().setPayload(
+              ByteString.copyFrom("a message for recipient", Charset.defaultCharset())));
+
+      Task newTask = client.createTask(CreateTaskRequest
+          .newBuilder()
+          .setParent(QueueName.of(projectId, location, queueName).toString())
+          .setTask(taskBuilder)
+          .build());
+      System.out.println("Task created: " + newTask.getName());
+    }
+  }
+  // [END cloud_tasks_create_task]
+
+  // [START cloud_tasks_lease_and_acknowledge_task]
+  private static void pullAndAckTask(String projectId, String queueName, String location) {
+    try (CloudTasksClient client = CloudTasksClient.create()) {
+      LeaseTasksRequest.Builder reqBuilder = LeaseTasksRequest.newBuilder()
+          .setParent(QueueName.of(projectId, location, queueName).toString())
+          .setLeaseDuration(Duration.newBuilder().setSeconds(600))
+          .setMaxTasks(1)
+          .setResponseView(Task.View.FULL);
+      LeaseTasksResponse response = client.leaseTasks(reqBuilder.build());
+      if (response.getTasksCount() == 0) {
+        System.out.println("No tasks found in queue.");
+        return;
+      }
+      Task task = response.getTasksList().get(0);
+      System.out.println("Leased task: " + task.getName());
+      client.acknowledgeTask(AcknowledgeTaskRequest
```

Comment:
What do you think about building the AcknowledgeTaskRequest before the function and passing it in? I think it increases readability and makes it more obvious what the function requires.

Reply:
Yeah that's fine.